From 641cb299d33557153925e67b0a61a960b6c35eb4 Mon Sep 17 00:00:00 2001
From: Markus Boehme <markus.boehme@chainguard.dev>
Date: Tue, 9 Dec 2025 20:58:19 +0000
Subject: [PATCH 5/5] vendor: go-selinux: use non-crypto rand for SELinux MCS
 categories

The vendored go-selinux package uses crypto/rand to determine two random
SELinux categories (in the range 0-1023), the combination of which are
supposed to be unique on a system, thereby separating e.g. processes
from one another. The random numbers do not need to be suitable for
cryptography for this purpose, so switch to math/rand/v2 to avoid having
to import the crypto package.

Signed-off-by: Markus Boehme <markus.boehme@chainguard.dev>
---
 .../opencontainers/selinux/go-selinux/selinux_linux.go | 10 +++-------
 1 file changed, 3 insertions(+), 7 deletions(-)

diff --git a/vendor/github.com/opencontainers/selinux/go-selinux/selinux_linux.go b/vendor/github.com/opencontainers/selinux/go-selinux/selinux_linux.go
index 6d7f8e27..1eda1dc8 100644
--- a/vendor/github.com/opencontainers/selinux/go-selinux/selinux_linux.go
+++ b/vendor/github.com/opencontainers/selinux/go-selinux/selinux_linux.go
@@ -3,13 +3,12 @@ package selinux
 import (
 	"bufio"
 	"bytes"
-	"crypto/rand"
-	"encoding/binary"
 	"errors"
 	"fmt"
 	"io"
 	"io/fs"
 	"math/big"
+	"math/rand/v2"
 	"os"
 	"os/user"
 	"path/filepath"
@@ -976,16 +975,13 @@ func intToMcs(id int, catRange uint32) string {
 
 func uniqMcs(catRange uint32) string {
 	var (
-		n      uint32
 		c1, c2 uint32
 		mcs    string
 	)
 
 	for {
-		_ = binary.Read(rand.Reader, binary.LittleEndian, &n)
-		c1 = n % catRange
-		_ = binary.Read(rand.Reader, binary.LittleEndian, &n)
-		c2 = n % catRange
+		c1 = rand.Uint32N(catRange)
+		c2 = rand.Uint32N(catRange)
 		if c1 == c2 {
 			continue
 		} else if c1 > c2 {
-- 
2.51.2

